crates/net/network/src/peers/manager.rs

```diff
@@ -740,6 +758,24 @@ impl PeerConnectionState {
     }
 }
 
+/// Represents the kind of peer
+#[derive(Debug, Clone, Copy, Default, Eq, PartialEq)]
+enum PeerKind {
+    /// Non-trusted peer.
+    #[default]
+    NonTrusted,
+    /// Trusted peer.
+    Trusted,
+}
+
+impl PeerKind {
+    /// Returns whether this peer is trusted
+    #[inline]
+    fn is_trusted(&self) -> bool {
+        matches!(self, Self::Trusted)
+    }
```

Comment:
this will be only called on the `Peer` itself so it'd like to move this to `impl Peer` instead so we can call it directly `peer.is_trusted()`